### ISSUE TITLE
Configured OnClickListeners are not working when horizontal calendar height changes

### ISSUE THREAD

**Issue by henriquenfaria** [kizitonwose/CalendarView#143]
Hi! first of all, thanks for this awesome library. 🥇 

I'm facing a weird problem when the horizontal calendar height changes (e.g. changing to a month that has 6 rows instead of 5): OnClickListeners don't work at all for the 6th row.

This problem can be reproduced by slightly changing the code from example 1. Basically, we need to replace the `ConstraintLayout` that wraps `CalendarView` with a simple `RelativeLayout` as shown below. Without this change, the last row is not even displayed unless a random day is clicked ( probably another issue related to `ConstraintLayouts`?):

```
<RelativeLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent">

        <com.kizitonwose.calendarview.CalendarView
            android:id="@+id/exOneCalendar"
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            android:layout_alignParentTop="true"
            app:cv_dayViewResource="@layout/example_1_calendar_day"
            app:cv_inDateStyle="allMonths"
            app:cv_orientation="horizontal"
            app:cv_outDateStyle="endOfRow"
            app:cv_scrollMode="paged" />

        <CheckBox
            android:id="@+id/weekModeCheckBox"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_alignParentBottom="true"
            android:layout_centerHorizontal="true"
            android:layout_marginBottom="40dp"
            android:buttonTint="@color/example_1_white"
            android:padding="2dp"
            android:text="@string/week_mode"
            android:textColor="@color/example_1_white"
            android:textSize="18sp" />

    </RelativeLayout>
``` 

Additionally, you should set the test device to **German**.  Here's a short GIF showing the problem:

![issue-resized](https://user-images.githubusercontent.com/1030977/78823402-e893fc80-79dc-11ea-886e-443f41d6ecd5.gif)

As you can see, the last row for March is not registering any clicks. It works again if you click on a day outside the last row.

I've been trying to debug this for the last few hours and I suspect this issue is caused by the `#ClipChildrenFix`fix in the codebase. Settings `clipChildren` to false makes the last row overflow outside its parent, because of this the previously registered OnClickListeners will fail to trigger.

If we remove the `clipChildren` fix the height is not updated and the last row is not shown.
Any ideas on how to proceed here? What I don't understand is why clicking on a random day helps.

**Comment by kizitonwose** [comment]
Hi, indeed the constraint layout issue is because in example 1, the calendar's bottom is not constrained to any view and with a height of `0dp`, the calendar can't properly measure itself. 

However, in a RelativeLayout, as shown in your example, the issue you're experiencing is now fixed in the new version [0.3.3](https://github.com/kizitonwose/CalendarView/releases/tag/0.3.3) 

The `#ClipChildrenFix` workaround was long due for removal has been rightfully removed and replaced with an appropriate fix for `wrap_content` mode.

**Comment by henriquenfaria** [comment]
The fix worked like a charm @kizitonwose 
Thanks a lot!
